### DISCUSSION TITLE
Populate Table component from SQL Server table

### DISCUSSION THREAD

**Discussion by Pieter3033** [sqlpage/SQLPage#108] [Q&A] [answered]
Hi all,

Is it possible to fully populate the Table component with a external table, in this case SQL Server table.

I am not sure if I missed some information on the website or does SQLPage not support direct table population.

I have tried the traditional INSERT INTO method, but I don't see how the syntaxes will match.

Then I attempted to use a SELECT method as the input, but I then realized the row_level_parameter only accepts one value.
See below for reference:
![image](https://github.com/lovasoa/SQLpage/assets/141907966/6b3e6177-eb52-4796-b1fa-4236157c202a)

![image](https://github.com/lovasoa/SQLpage/assets/141907966/a23c98ee-8ea1-4c82-959d-c285a7370071)

I then attempted to use a while loop to populate one by one, with the discovery that I believe SQLPage does not have While loop support. /**Please correct me if I'm wrong**/
See below for reference:
![image](https://github.com/lovasoa/SQLpage/assets/141907966/3e51cfae-0a2e-4d29-87ea-da6ff44eba98)

![image](https://github.com/lovasoa/SQLpage/assets/141907966/75c8e1cf-8da8-46bc-aa42-ca8435c461cd)

**Comment by Pieter3033** [comment]
Now I realize that the above loop will create a table component for the ammounts of id's, but when I attempt to implement it inside of the row_level_parameter I can't see how the syntaxes will match

**Comment by lovasoa** [comment, marked as answer]
You are getting into very complicated things. You just need to select from your table directly.

```sql
select 'table' as component;
select Username from JC_User_Info;
```

That's all. That will populate the table component with a single column named Username and the values from the `JC_User_Info` table.

**Reply by lovasoa** [reply]
After opening the `table` component, you can make any arbitrarily complex SQL query, with joins, aggregate, subqueries, and anything you want, and the results will be displayed as a table in the browser.

**Reply by lovasoa** [reply]
The documentation needs to be self-contained, so it only gives very simple examples with `select` statements with static data and without a `from` clause, but you are not limited to this simplistic case when building your application.

You can have a look at [the official repository of examples](https://github.com/lovasoa/SQLpage/tree/main/examples) to see more realistic usage of the components, where data is queried from actual databases.

**Reply by Pieter3033** [reply]
Oh my...👀

Thank you for the assistance. It is appreciated.
I will definitely have to thoroughly examine the examples.
